added a patch to prevent segfault when Ctrl+R is hit on a malformed group of
messages (see upstream bug http://bugs.mutt.org/3335)

--- a/flags.c
+++ b/flags.c
@@ -303,7 +303,10 @@
 int mutt_thread_set_flag (HEADER *hdr, int flag, int bf, int subthread)
 {
   THREAD *start, *cur = hdr->thread;
-  
+
+  if ( !hdr->thread )
+    return (-1);
+
   if ((Sort & SORT_MASK) != SORT_THREADS)
   {
     mutt_error _("Threading is not enabled.");
